sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapBuilder.scala

```diff
@@ -63,6 +68,11 @@ class ArrayBasedMapBuilder(keyType: DataType, valueType: DataType) extends Seria
       keys.append(key)
       values.append(value)
     } else {
+      if (!allowDuplicatedMapKey) {
+        throw new RuntimeException(s"Duplicate map key $key was founded, please set " +
```

Comment:
We shouldn't recommend users to set legacy config to fix a problem. We should ask them to check the input data to see why there are duplicated keys. If they want to remove duplicated keys, tell them that they can set the legacy config which uses last wins policy.

Reply:
Thanks, done in 905ae96.